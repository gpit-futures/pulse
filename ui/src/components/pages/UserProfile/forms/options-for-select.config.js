const optionsForThemesField = [{
  value: 'green',
  title: 'Green Theme',
},
{
  value: 'purple',
  title: 'Purple Theme',
},
{
  value: 'gray',
  title: 'Gray Theme',
}];

const optionsForGenderField = [{
  value: 'female',
  title: 'Female',
}, {
  value: 'male',
  title: 'Male',
}];

const optionsForCountryField = [{
  value: 'usa',
  title: 'USA',
}, {
  value: 'country1',
  title: 'Country 1',
}, {
  value: 'country2',
  title: 'Country 2',
}, {
  value: 'country3',
  title: 'Country 3',
}, {
  value: 'country4',
  title: 'Country 4',
}, {
  value: 'country5',
  title: 'Country 5',
}];

export { optionsForThemesField, optionsForGenderField, optionsForCountryField }
