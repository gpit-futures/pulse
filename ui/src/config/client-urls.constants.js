import { themeConfigs } from '../themes.config';

export const clientUrls = {
  ROOT: '/',
  CHARTS: '/charts',
  PATIENTS: '/patients',
  PATIENTS_FULL_DETAILS: '/patients-full-details',
  SEARCH_REPORT: '/search-report',
  USER_PROFILE: '/profile',
  PATIENTS_SUMMARY: 'patients-summary',
  RESULTS: 'results',
  DIAGNOSES: 'diagnoses',
  ALLERGIES: 'allergies',
  CONTACTS: 'contacts',
  CLINICAL_NOTES: 'clinicalNotes',
  PERSONAL_NOTES: 'personalNotes',
  VACCINATIONS: 'vaccinations',
  MEDICATIONS: 'medications',
  GENERIC_PLUGIN: 'genericPlugin',
  PROCEDURES: 'procedures',
  EVENTS: 'events',
  TEST_RESULTS: 'results',
  REFERRALS: 'referrals',
  ORDERS: 'orders',
  MDTS: 'mdt',
  DRAWINGS: 'drawings',
  VITALS: 'vitals',
  DOCUMENTS: 'documents',
  IMAGES: 'images',
  TRANSFERS_OF_CARE: 'transfer-of-care',
  TOP_THREE_THINGS: 'topThreeThings',
  CLINICAL_STATEMENTS: 'clinicalStatements',
  PROMS: 'proms',
  FEEDS: 'feeds',
};

const pluginsPages = {
  'allergies': {
    breadcrumbs: [{
      title: 'Allergies',
      state: '/allergies',
    }],
  },
  'diagnoses': {
    breadcrumbs: [{
      title: 'Problems / Diagnoses',
      state: '/diagnoses',
    }],
  },
  'clinicalNotes': {
    breadcrumbs: [{
      title: 'Clinical Notes',
      state: '/clinicalNotes',
    }],
  },
  'personalNotes': {
    breadcrumbs: [{
      title: 'Personal Notes',
      state: '/personalNotes',
    }],
  },
  'contacts': {
    breadcrumbs: [{
      title: 'Contacts',
      state: '/contacts',
    }],
  },
  'vaccinations': {
    breadcrumbs: [{
      title: 'Vaccinations',
      state: '/vaccinations',
    }],
  },
  'genericPlugin': {
    breadcrumbs: [{
      title: 'Generic Plugin',
      state: '/genericPlugin',
    }],
  },
  'medications': {
    breadcrumbs: [{
      title: 'Medications',
      state: '/medications',
    }],
  },
  'procedures': {
    breadcrumbs: [{
      title: 'Procedures',
      state: '/procedures',
    }],
  },
  'events': {
    breadcrumbs: [{
      title: 'Events',
      state: '/events',
    }],
  },
  'results': {
    breadcrumbs: [{
      title: 'Test Results',
      state: '/results',
    }],
  },
  'referrals': {
    breadcrumbs: [{
      title: 'Referrals',
      state: '/referrals',
    }],
  },
  'orders': {
    breadcrumbs: [{
      title: 'Orders',
      state: '/orders',
    }],
  },
  'mdt': {
    breadcrumbs: [{
      title: 'MDT',
      state: '/mdt',
    }],
  },
  'drawings': {
    breadcrumbs: [{
      title: 'Drawings',
      state: '/drawings',
    }],
  },
  'vitals': {
    breadcrumbs: [{
      title: 'Vitals - NEWS',
      state: '/vitals',
    }],
  },
  'documents': {
    breadcrumbs: [{
      title: 'Documents',
      state: '/documents',
    }],
  },
  'images': {
    breadcrumbs: [{
      title: 'Images',
      state: '/images',
    }],
  },
  'transfer-of-care': {
    breadcrumbs: [{
      title: 'Transfers Of Care',
      state: '/transfers-of-care',
    }],
  },
  'topThreeThings': {
    breadcrumbs: [{
      title: 'Top 3 Things',
      state: '/topThreeThings',
    }],
  },
  'clinicalStatements': {
    breadcrumbs: [{
      title: 'Clinical Statements',
      state: '/clinicalStatements',
    }],
  },
  'proms': {
    breadcrumbs: [{
      title: 'PROMs',
      state: '/proms',
    }],
  },
};


const addPluginsPagesToLists = (list, breadcrumbsBefore) => {
  for (const nameOfPage in pluginsPages) {
    const breadcrumbs = breadcrumbsBefore.concat(pluginsPages[nameOfPage].breadcrumbs);
    list[nameOfPage] = { breadcrumbs };
  }
};

export const mainPagesTitles = {
  '/': {
    breadcrumbs: [{
      title: 'System Dashboard',
      state: '/',
    }],
    headerTitle: 'System Dashboard',
  },
  'charts': {
    breadcrumbs: [{
      title: 'System Dashboard',
      state: '/charts',
    }],
    headerTitle: 'System Dashboard',
  },
  'patients': {
    breadcrumbs: [{
      title: 'Home',
      state: '/',
    }, {
      title: 'Patient Listings',
      state: '/patients',
    }],
    headerTitle: 'Patients Lists',
  },
  'profile': {
    breadcrumbs: [{
      title: 'Patient Information',
      state: '/profile',
    }],
    headerTitle: 'Personal Information',
  },
  'search-report': {
    breadcrumbs: [{
      title: 'Home',
      state: '/',
    }, {
      title: 'Search Report',
      state: '/search-report',
    }],
    headerTitle: 'System Dashboard',
  },
  'patients-summary': {
    breadcrumbs: [{
      title: 'Patient Listings',
      state: '/patients',
    }, {
      title: themeConfigs.patientsSummaryPageName,
      state: '/patients-summary',
    }],
  },
  'patients-full-details': {
    breadcrumbs: [],
    headerTitle: 'Search results',
  },
};
addPluginsPagesToLists(mainPagesTitles, [{
  title: 'Patient Listings',
  state: '/patients',
}, {
  title: themeConfigs.patientsSummaryPageName,
  state: '/patients-summary',
}]);

export const mainPagesTitlesForPatients = {
  '/': {
    breadcrumbs: [{
      title: 'Patient Summary',
      state: '/patients-summary',
    }],
    headerTitle: 'Patient Summary',
  },

  'profile': {
    breadcrumbs: [{
      title: 'Patient Information',
      state: '/profile',
    }],
    headerTitle: 'Personal Information',
  },
  'patients-summary': {
    breadcrumbs: [{
      title: themeConfigs.patientsSummaryPageName,
      state: '/patients-summary',
    }],
  },
};
addPluginsPagesToLists(mainPagesTitlesForPatients, [{
  title: themeConfigs.patientsSummaryPageName,
  state: '/patients-summary',
}]);
