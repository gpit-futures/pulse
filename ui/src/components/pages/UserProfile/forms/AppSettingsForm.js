import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Field, reduxForm } from 'redux-form'
import { Row, Col } from 'react-bootstrap';

import ValidatedInput from '../../../form-fields/ValidatedInputFormGroup';
import FileInput from '../../../form-fields/FileInput';
import { optionsForThemesField } from './options-for-select.config';
import Select from '../../../form-fields/SelectFormGroup';
import { validateAppSettingsForm } from '../forms.validation';
import { valuesSettingsForm, valuesSettingsFormLabels } from '../forms.config';
import { setTheme } from '../../../../ducks/set-theme.duck';

@reduxForm({
  form: 'appSettingsFormSelector',
  validate: validateAppSettingsForm,
})
export default class AppSettingsForm extends PureComponent {
  static propTypes = {
    dispatch: PropTypes.func,
    initialize: PropTypes.func,
  };

  componentDidMount() {
    const { patientsInfo, initialize } = this.props;
    initialize(this.defaultValuesForm(patientsInfo));
  }

  defaultValuesForm(value) {
    const defaultFormValues = {
      [valuesSettingsForm.APP_TITLE]: value.title,
      [valuesSettingsForm.LOGO_PATH]: value.logoB64,
      [valuesSettingsForm.SELECT_THEME]: value.themeColor,
      [valuesSettingsForm.BROWSER_TITLE]: value.browserTitle,
    };

    return defaultFormValues;
  }

 setThemeHook = dispatch => (theme) => {
   dispatch(setTheme(theme));
   return theme;
 };

 render() {
   const { dispatch } = this.props;

   return (
     <div className="panel-body-inner">
       <form name="appSettingsForm" className="form">
         <div className="form-group-wrapper">
           <Row>
             <Col xs={12} md={6}>
               <Row>
                 <Col md={11}>
                   <Field
                     label={valuesSettingsFormLabels.APP_TITLE}
                     name={valuesSettingsForm.APP_TITLE}
                     type="text"
                     placeholder=""
                     component={ValidatedInput}
                   />
                   <Field
                     label={valuesSettingsFormLabels.LOGO_PATH}
                     name={valuesSettingsForm.LOGO_PATH}
                     component={FileInput}
                     id={valuesSettingsForm.LOGO_PATH}
                   />
                   <Field
                     label={valuesSettingsFormLabels.SELECT_THEME}
                     name={valuesSettingsForm.SELECT_THEME}
                     component={Select}
                     options={optionsForThemesField}
                     //here we're using normalize hook for dispatching action to set app theme
                     normalize={this.setThemeHook(dispatch)}
                   />
                   <Field
                     label={valuesSettingsFormLabels.BROWSER_TITLE}
                     name={valuesSettingsForm.BROWSER_TITLE}
                     type="text"
                     placeholder=""
                     component={ValidatedInput}
                   />
                 </Col>
               </Row>
             </Col>
           </Row>
         </div>
       </form>
     </div>)
 }
}
