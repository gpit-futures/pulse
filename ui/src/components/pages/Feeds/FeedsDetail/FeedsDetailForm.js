import React, { PureComponent } from 'react';
import { Field, reduxForm } from 'redux-form'

import ValidatedInput from '../../../form-fields/ValidatedInputFormGroup';
import ValidatedTextareaFormGroup from '../../../form-fields/ValidatedTextareaFormGroup';
import DateInput from '../../../form-fields/DateInput';
import { valuesNames, valuesLabels } from '../forms.config';
import { validateForm } from '../forms.validation';

@reduxForm({
  form: 'feedsPanelFormSelector',
  validate: validateForm,
})
export default class FeedsDetailForm extends PureComponent {
  componentDidMount() {
    const { detail, initialize } = this.props;
    initialize(this.defaultValuesForm(detail));
  }
  defaultValuesForm(value) {
    const defaultFormValues = {
      [valuesNames.NAME]: value[valuesNames.NAME],
      [valuesNames.LANDING_PAGE_URL]: value[valuesNames.LANDING_PAGE_URL],
      [valuesNames.RSS_FEED_URL]: value[valuesNames.RSS_FEED_URL],
      [valuesNames.AUTHOR]: value[valuesNames.AUTHOR],
    };

    return defaultFormValues;
  }
  render() {
    const { detail, isSubmit } = this.props;
    return (
      <div className="panel-body-inner">
        <form name="feedsPanelForm" className="form">
          <div className="form-group-wrapper">
            <div className="row-expand">
              <div className="col-expand-left">
                <Field
                  label={valuesLabels.NAME}
                  name={valuesNames.NAME}
                  id={valuesNames.NAME}
                  type="text"
                  placeholder=""
                  component={ValidatedInput}
                  props={{ isSubmit }}
                />
              </div>
            </div>
            <div className="row-expand">
              <div className="col-expand-left">
                <Field
                  label={valuesLabels.LANDING_PAGE_URL}
                  name={valuesNames.LANDING_PAGE_URL}
                  id={valuesNames.LANDING_PAGE_URL}
                  component={ValidatedTextareaFormGroup}
                  props={{ isSubmit }}
                />
              </div>
              <div className="col-expand-right">
                <Field
                  label={valuesLabels.RSS_FEED_URL}
                  name={valuesNames.RSS_FEED_URL}
                  id={valuesNames.RSS_FEED_URL}
                  component={ValidatedTextareaFormGroup}
                  props={{ isSubmit }}
                />
              </div>
            </div>
            <div className="row-expand">
              <div className="col-expand-left">
                <Field
                  label={valuesLabels.AUTHOR}
                  name={valuesNames.AUTHOR}
                  id={valuesNames.AUTHOR}
                  component={ValidatedInput}
                  props={{ disabled: true }}
                />
              </div>
              <div className="col-expand-right">
                <Field
                  label={valuesLabels.DATE_CREATED}
                  name={valuesNames.DATE_CREATED}
                  id={valuesNames.DATE_CREATED}
                  component={DateInput}
                  props={{ disabled: true, value: detail[valuesNames.DATE_CREATED], format: 'DD-MMM-YYYY' }}
                />
              </div>
            </div>
          </div>
        </form>
      </div>)
  }
}
