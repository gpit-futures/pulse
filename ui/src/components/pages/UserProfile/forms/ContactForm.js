import React, { PureComponent } from 'react';
import { Field, reduxForm } from 'redux-form'
import { Row, Col } from 'react-bootstrap';

import ValidatedInput from '../../../form-fields/ValidatedInputFormGroup';
import { optionsForCountryField } from './options-for-select.config';
import Select from '../../../form-fields/SelectFormGroup';
import { validateContactForm } from '../forms.validation';
import { valuesContactForm, valuesContactFormLabels } from '../forms.config';
import { defaultContactFormValues } from './default-values.config';

@reduxForm({
  form: 'contactFormSelector',
  validate: validateContactForm,
})
export default class ContactForm extends PureComponent {
  componentDidMount() {
    this.props.initialize(defaultContactFormValues);
  }

  render() {
    return (
      <div className="panel-body-inner">
        <form name="contactForm" className="form">
          <div className="form-group-wrapper">
            <Row>
              <Col xs={12} md={6}>
                <Row>
                  <Col md={11}>
                    <Field
                      label={valuesContactFormLabels.ADDRESS}
                      name={valuesContactForm.ADDRESS}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label={valuesContactFormLabels.CITY}
                      name={valuesContactForm.CITY}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label={valuesContactFormLabels.STATE}
                      name={valuesContactForm.STATE}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label={valuesContactFormLabels.POSTAL_CODE}
                      name={valuesContactForm.POSTAL_CODE}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label={valuesContactFormLabels.SELECT_COUNTRY}
                      name={valuesContactForm.SELECT_COUNTRY}
                      placeholder=""
                      component={Select}
                      options={optionsForCountryField}
                    />
                  </Col>
                </Row>
              </Col>
              <Col xs={12} md={6}>
                <Row>
                  <Col md={11} mdOffset={1}>
                    <Field
                      label={valuesContactFormLabels.PHONE}
                      name={valuesContactForm.PHONE}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label={valuesContactFormLabels.EMAIL}
                      name={valuesContactForm.EMAIL}
                      placeholder=""
                      component={ValidatedInput}
                    />
                  </Col>
                </Row>
              </Col>
            </Row>
          </div>
        </form>
      </div>)
  }
}
