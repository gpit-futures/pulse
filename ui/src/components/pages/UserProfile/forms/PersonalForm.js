import React, { PureComponent } from 'react';
import { Field, reduxForm } from 'redux-form'
import { Row, Col } from 'react-bootstrap';

import ValidatedInput from '../../../form-fields/ValidatedInputFormGroup';
import DateInput from '../../../form-fields/DateInput';
import { optionsForGenderField } from './options-for-select.config';
import Select from '../../../form-fields/SelectFormGroup';
import { validatePersonalForm } from '../forms.validation';
import { valuesPersonalForm, valuesPersonalFormLabels } from '../forms.config';
import { defaultPersonalFormValues } from './default-values.config';

@reduxForm({
  form: 'personalFormSelector',
  validate: validatePersonalForm,
})
export default class PersonalForm extends PureComponent {
  componentDidMount() {
    this.props.initialize(defaultPersonalFormValues);
  }

  render() {
    return (
      <div className="panel-body-inner">
        <form name="personalForm" className="form">
          <div className="form-group-wrapper">
            <Row>
              <Col xs={12} md={6}>
                <Row>
                  <Col md={11}>
                    <Field
                      label={valuesPersonalFormLabels.FIRST_NAME}
                      name={valuesPersonalForm.FIRST_NAME}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label={valuesPersonalFormLabels.LAST_NAME}
                      name={valuesPersonalForm.LAST_NAME}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label={valuesPersonalFormLabels.NHS_NUMBER}
                      name={valuesPersonalForm.NHS_NUMBER}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                  </Col>
                </Row>
              </Col>
              <Col xs={12} md={6}>
                <Row>
                  <Col md={11} mdOffset={1}>
                    <Field
                      label={valuesPersonalFormLabels.DATE_OF_BIRTH}
                      name={valuesPersonalForm.DATE_OF_BIRTH}
                      type="text"
                      placeholder=""
                      component={DateInput}
                      props={{ format: 'DD-MMM-YYYY' }}
                    />
                    <Field
                      label={valuesPersonalFormLabels.SELECT_GENDER}
                      name={valuesPersonalForm.SELECT_GENDER}
                      placeholder=""
                      component={Select}
                      options={optionsForGenderField}
                    />
                    <Field
                      label={valuesPersonalFormLabels.DOCTOR}
                      name={valuesPersonalForm.DOCTOR}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                  </Col>
                </Row>
              </Col>
            </Row>
          </div>
        </form>
      </div>)
  }
}
