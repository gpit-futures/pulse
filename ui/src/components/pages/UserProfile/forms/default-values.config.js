import { valuesSettingsForm, valuesPersonalForm, valuesContactForm } from '../forms.config';

const defaultPersonalFormValues = {
  [valuesPersonalForm.FIRST_NAME]: 'Bob',
  [valuesPersonalForm.LAST_NAME]: 'Smith',
  [valuesPersonalForm.SELECT_GENDER]: 'female',
  [valuesPersonalForm.DOCTOR]: 'Dr Emma Huston',
  [valuesPersonalForm.DATE_OF_BIRTH]: 0,
};

const defaultContactFormValues = {
  [valuesContactForm.ADDRESS]: '6801 Tellus Street',
  [valuesContactForm.CITY]: 'Westmorland',
  [valuesContactForm.STATE]: 'Westmorland',
  [valuesContactForm.POSTAL_CODE]: 'Box 306',
  [valuesContactForm.PHONE]: '07624 647524',
  [valuesContactForm.EMAIL]: 'bob.smith@gmail.com',
};

const defaultAppSettingsFormValues = {
  [valuesSettingsForm.APP_TITLE]: 'should be fetched from server',
  [valuesSettingsForm.LOGO_PATH]: 'should be fetched from server',
  [valuesSettingsForm.SELECT_THEME]: 'themeDefault',
  [valuesSettingsForm.BROWSER_TITLE]: 'should be fetched from server',
};

export { defaultPersonalFormValues, defaultContactFormValues, defaultAppSettingsFormValues }
