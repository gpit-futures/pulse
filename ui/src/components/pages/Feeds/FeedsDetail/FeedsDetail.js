import React, { PureComponent } from 'react';

import PluginDetailPanel from '../../../plugin-page-component/PluginDetailPanel'
import FeedsDetailForm from './FeedsDetailForm'
import { getDDMMMYYYY } from '../../../../utils/time-helpers.utils';
import { valuesNames, valuesLabels } from '../forms.config';

const FEEDS_PANEL = 'feedsPanel';

export default class FeedsDetail extends PureComponent {
  render() {
    const { onExpand, openedPanel, expandedPanel, currentPanel, onEdit, editedPanel, onCancel, onSaveSettings, feedFormValues, isSubmit } = this.props;
    let { detail } = this.props;
    detail = detail || {};
    const dateCreated = getDDMMMYYYY(detail[valuesNames.DATE_CREATED]);

    return (
      <div className="section-detail">
        <div className="panel-group accordion">
          {(expandedPanel === FEEDS_PANEL || expandedPanel === 'all') && !editedPanel[FEEDS_PANEL] ? <PluginDetailPanel
            onExpand={onExpand}
            name={FEEDS_PANEL}
            title="Feed"
            isOpen={openedPanel === FEEDS_PANEL}
            currentPanel={currentPanel}
            onEdit={onEdit}
            editedPanel={editedPanel}
            onCancel={onCancel}
            onSaveSettings={onSaveSettings}
            formValues={feedFormValues}
            isBtnShowPanel={false}
          >
            <div className="panel-body-inner">
              <div className="form">
                <div className="form-group-wrapper">
                  <div>
                    <div className="row-expand">
                      <div className="col-expand-left">
                        <div className="form-group">
                          <label className="control-label">{valuesLabels.NAME}</label>
                          <div className="form-control-static">{detail[valuesNames.NAME]}</div>
                        </div>
                      </div>
                    </div>
                    <div className="row-expand">
                      <div className="col-expand-left">
                        <div className="form-group">
                          <label className="control-label">{valuesLabels.LANDING_PAGE_URL}</label>
                          <div className="form-control-static">{detail[valuesNames.LANDING_PAGE_URL]}</div>
                        </div>
                      </div>
                      <div className="col-expand-right">
                        <div className="form-group">
                          <label className="control-label">{valuesLabels.RSS_FEED_URL}</label>
                          <div className="form-control-static">{detail[valuesNames.RSS_FEED_URL]}</div>
                        </div>
                      </div>
                    </div>
                    <div className="row-expand">
                      <div className="col-expand-left">
                        <div className="form-group">
                          <label className="control-label">{valuesLabels.AUTHOR}</label>
                          <div className="form-control-static">{detail[valuesNames.AUTHOR]}</div>
                        </div>
                      </div>
                      <div className="col-expand-right">
                        <div className="form-group">
                          <label className="control-label">{valuesLabels.DATE_CREATED}</label>
                          <div className="form-control-static">{dateCreated}</div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </PluginDetailPanel> : null}
          {(expandedPanel === FEEDS_PANEL || expandedPanel === 'all') && editedPanel[FEEDS_PANEL] ? <PluginDetailPanel
            onExpand={onExpand}
            name={FEEDS_PANEL}
            title="Edit Feed"
            isOpen={openedPanel === FEEDS_PANEL}
            currentPanel={currentPanel}
            onEdit={onEdit}
            editedPanel={editedPanel}
            onCancel={onCancel}
            onSaveSettings={onSaveSettings}
            formValues={feedFormValues}
            isBtnShowPanel={false}
          >
            <FeedsDetailForm
              detail={detail}
              isSubmit={isSubmit}
            />
          </PluginDetailPanel> : null}
        </div>
      </div>
    )
  }
}
