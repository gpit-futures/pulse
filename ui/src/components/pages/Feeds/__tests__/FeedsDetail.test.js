import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';

import FeedsDetail from '../FeedsDetail/FeedsDetail';
import { valuesNames, valuesLabels } from '../forms.config';
import { getDDMMMYYYY } from '../../../../utils/time-helpers.utils';

Enzyme.configure({ adapter: new Adapter() });

const FEEDS_PANEL = 'feedsPanel';
const FEEDS_DETAIL_TITLE = 'Feed';
const FEEDS_DETAIL_EDIT_TITLE = 'Edit Feed';

const testProps = {
  onExpand: () => {},
  onEdit: () => {},
  onCancel: () => {},
  onSaveSettings: () => {},
  openedPanel: {},
  expandedPanel: 'all',
  currentPanel: 'currentPanel',
  editedPanel: 'editedPanel',
  feedFormValues: {},
};

const detail = {
  [valuesNames.NAME]: 'BBC Health',
  [valuesNames.LANDING_PAGE_URL]: 'http://www.bbc.co.uk/news/health',
  [valuesNames.RSS_FEED_URL]: 'http://feeds.bbci.co.uk/news/health/rss.xml?edition=uk#',
  [valuesNames.SOURCE_ID]: 'testSourceID1',
  [valuesNames.DATE_CREATED]: 1482170593395,
  [valuesNames.AUTHOR]: 'bob.smith@gmail.com',
};

const CONVERT_DATE = getDDMMMYYYY(detail[valuesNames.DATE_CREATED]);

describe('Component <FeedsDetail />', () => {
  it('should renders with all props correctly', () => {
    const component = shallow(
      <FeedsDetail
        detail={detail}
        onExpand={testProps.onExpand}
        onEdit={testProps.onEdit}
        onCancel={testProps.onCancel}
        onSaveSettings={testProps.onSaveSettings}
        openedPanel={testProps.openedPanel}
        expandedPanel={testProps.expandedPanel}
        currentPanel={testProps.currentPanel}
        editedPanel={testProps.editedPanel}
        feedFormValues={testProps.feedFormValues}
        isSubmit={false}
      />
    );

    expect(component).toMatchSnapshot();

    // Testing component when detail filled object, expandedPanel is all, and panel not edited
    expect(component.instance().props.detail).toEqual(detail);
    expect(component.instance().props.onExpand).toEqual(testProps.onExpand);
    expect(component.instance().props.onEdit).toEqual(testProps.onEdit);
    expect(component.instance().props.onCancel).toEqual(testProps.onCancel);
    expect(component.instance().props.onSaveSettings).toEqual(testProps.onSaveSettings);
    expect(component.instance().props.openedPanel).toEqual(testProps.openedPanel);
    expect(component.instance().props.expandedPanel).toEqual(testProps.expandedPanel);
    expect(component.instance().props.currentPanel).toEqual(testProps.currentPanel);
    expect(component.instance().props.editedPanel).toEqual(testProps.editedPanel);
    expect(component.instance().props.feedFormValues).toEqual(testProps.feedFormValues);
    expect(component.instance().props.isSubmit).toEqual(false);

    expect(component.find('.section-detail')).toHaveLength(1);
    expect(component.find('.form')).toHaveLength(1);
    expect(component.find('PluginDetailPanel')).toHaveLength(1);

    expect(component.find('PluginDetailPanel').props().currentPanel).toEqual(testProps.currentPanel);
    expect(component.find('PluginDetailPanel').props().editedPanel).toEqual(testProps.editedPanel);
    expect(component.find('PluginDetailPanel').props().isBtnShowPanel).toEqual(false);
    expect(component.find('PluginDetailPanel').props().isOpen).toEqual(testProps.editedPanel === FEEDS_PANEL);
    expect(component.find('PluginDetailPanel').props().name).toEqual(FEEDS_PANEL);
    expect(component.find('PluginDetailPanel').props().onCancel).toEqual(testProps.onCancel);
    expect(component.find('PluginDetailPanel').props().onEdit).toEqual(testProps.onEdit);
    expect(component.find('PluginDetailPanel').props().onExpand).toEqual(testProps.onExpand);
    expect(component.find('PluginDetailPanel').props().onSaveSettings).toEqual(testProps.onSaveSettings);
    expect(component.find('PluginDetailPanel').props().title).toEqual(FEEDS_DETAIL_TITLE);


    expect(component.find('.control-label').at(0).text()).toEqual(valuesLabels.NAME);
    expect(component.find('.control-label').at(1).text()).toEqual(valuesLabels.LANDING_PAGE_URL);
    expect(component.find('.control-label').at(2).text()).toEqual(valuesLabels.RSS_FEED_URL);
    expect(component.find('.control-label').at(3).text()).toEqual(valuesLabels.AUTHOR);
    expect(component.find('.control-label').at(4).text()).toEqual(valuesLabels.DATE_CREATED);

    expect(component.find('.form-control-static').at(0).text()).toEqual(detail[valuesNames.NAME]);
    expect(component.find('.form-control-static').at(1).text()).toEqual(detail[valuesNames.LANDING_PAGE_URL]);
    expect(component.find('.form-control-static').at(2).text()).toEqual(detail[valuesNames.RSS_FEED_URL]);
    expect(component.find('.form-control-static').at(3).text()).toEqual(detail[valuesNames.AUTHOR]);
    expect(component.find('.form-control-static').at(4).text()).toEqual(CONVERT_DATE);
  });

  it('should renders correctly with different state of props', () => {
    const component = shallow(
      <FeedsDetail
        expandedPanel={'all'}
        editedPanel={{ [FEEDS_PANEL]: true }}
        openedPanel={{ [FEEDS_PANEL]: true }}
      />
    );
    expect(component).toMatchSnapshot();

    // Testing component when detail filled object, expandedPanel is all, and panel is edited
    component.setProps({ detail });
    expect(component).toMatchSnapshot();

    expect(component.find('PluginDetailPanel')).toHaveLength(1);
    expect(component.find('ReduxForm')).toHaveLength(1);
    expect(component.find('PluginDetailPanel').props().title).toEqual(FEEDS_DETAIL_EDIT_TITLE);
  });
});

