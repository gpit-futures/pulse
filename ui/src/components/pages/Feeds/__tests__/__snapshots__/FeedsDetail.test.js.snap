// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <FeedsDetail /> should renders correctly with different state of props 1`] = `
<div
  className="section-detail"
>
  <div
    className="panel-group accordion"
  >
    <PluginDetailPanel
      editedPanel={
        Object {
          "feedsPanel": true,
        }
      }
      isBtnShowPanel={false}
      isOpen={false}
      isShowControlPanel={true}
      name="feedsPanel"
      title="Edit Feed"
    >
      <ReduxForm
        detail={Object {}}
      />
    </PluginDetailPanel>
  </div>
</div>
`;

exports[`Component <FeedsDetail /> should renders correctly with different state of props 2`] = `
<div
  className="section-detail"
>
  <div
    className="panel-group accordion"
  >
    <PluginDetailPanel
      editedPanel={
        Object {
          "feedsPanel": true,
        }
      }
      isBtnShowPanel={false}
      isOpen={false}
      isShowControlPanel={true}
      name="feedsPanel"
      title="Edit Feed"
    >
      <ReduxForm
        detail={
          Object {
            "author": "bob.smith@gmail.com",
            "dateCreated": 1482170593395,
            "landingPageUrl": "http://www.bbc.co.uk/news/health",
            "name": "BBC Health",
            "rssFeedUrl": "http://feeds.bbci.co.uk/news/health/rss.xml?edition=uk#",
            "sourceId": "testSourceID1",
          }
        }
      />
    </PluginDetailPanel>
  </div>
</div>
`;

exports[`Component <FeedsDetail /> should renders with all props correctly 1`] = `
<div
  className="section-detail"
>
  <div
    className="panel-group accordion"
  >
    <PluginDetailPanel
      currentPanel="currentPanel"
      editedPanel="editedPanel"
      formValues={Object {}}
      isBtnShowPanel={false}
      isOpen={false}
      isShowControlPanel={true}
      name="feedsPanel"
      onCancel={[Function]}
      onEdit={[Function]}
      onExpand={[Function]}
      onSaveSettings={[Function]}
      title="Feed"
    >
      <div
        className="panel-body-inner"
      >
        <div
          className="form"
        >
          <div
            className="form-group-wrapper"
          >
            <div>
              <div
                className="row-expand"
              >
                <div
                  className="col-expand-left"
                >
                  <div
                    className="form-group"
                  >
                    <label
                      className="control-label"
                    >
                      Name
                    </label>
                    <div
                      className="form-control-static"
                    >
                      BBC Health
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="row-expand"
              >
                <div
                  className="col-expand-left"
                >
                  <div
                    className="form-group"
                  >
                    <label
                      className="control-label"
                    >
                      Landing page URL
                    </label>
                    <div
                      className="form-control-static"
                    >
                      http://www.bbc.co.uk/news/health
                    </div>
                  </div>
                </div>
                <div
                  className="col-expand-right"
                >
                  <div
                    className="form-group"
                  >
                    <label
                      className="control-label"
                    >
                      RSS feed URL
                    </label>
                    <div
                      className="form-control-static"
                    >
                      http://feeds.bbci.co.uk/news/health/rss.xml?edition=uk#
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="row-expand"
              >
                <div
                  className="col-expand-left"
                >
                  <div
                    className="form-group"
                  >
                    <label
                      className="control-label"
                    >
                      Author
                    </label>
                    <div
                      className="form-control-static"
                    >
                      bob.smith@gmail.com
                    </div>
                  </div>
                </div>
                <div
                  className="col-expand-right"
                >
                  <div
                    className="form-group"
                  >
                    <label
                      className="control-label"
                    >
                      Date
                    </label>
                    <div
                      className="form-control-static"
                    >
                      19-Dec-2016
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </PluginDetailPanel>
  </div>
</div>
`;
